File: BE/coppy/shop-interior/src/main/java/com/example/shopinterior/entity/oder/Oder.java
package com.example.shopinterior.entity.oder;

import com.example.shopinterior.entity.account.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Oder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idOder;
    private String flagDelete;
    private String orderDate;
    private String paymentMethod;
    @JsonManagedReference
    @ManyToOne
    private User user;
    @JsonBackReference
    @OneToMany(mappedBy = "oder")
    private Set<PurchaseHistory> purchaseHistorySet;
}

File: FE/shop-interior/shop-interior/src/app/entity/oder.ts
import {User} from "./user";
import {PurchaseHistory} from "./purchase-history";

export interface Oder {
  idOder?: number;
  code?: string;
  deliveryAddress?: string;
  deliverPhone?: string;
  orderDate?: string;
  paymentMethod?: boolean;
  userDto?: User;
  purchaseHistorySet?: PurchaseHistory[];
  orderValue?: number;

}
